### PR TITLE
Add hospital capacity data and downstream effects

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR adds spare hospital capacity data to the package taken from the [DAEDALUS data repository](https://github.com/patcatgit/daedalus_data), and two downstream effects.

1. Spare hospital capacity data raw data file has been added to `inst/extdata`. A processing script to generate an intermediate dataset has been added to `data-raw/`. The number of countries with available data has been reduced to 41.  `<daedalus_country>` class objects now include spare capacity data. Added tests check for hospital capacity related effects, but I haven't found a good test for the change in mortality rate when capacity is exceeded.

## Effects

- A call to `daedalus()` now takes the `response_threshold` from the `country`-specific hospital capacity by default. Users can still pass a custom value to override the package data.

- The model's mortality rate `omega` is increased by `1.6x` when the total number of hospitalisations exceed the spare hospital capacity. This is implemented using a switch `hosp_switch` mutable parameter, and the switch activation and termination is rolled into the event functions that control closures.

### CODE REVIEW

File: R/closure.R [hunk 1]
```diff
@@ -18,14 +18,18 @@ make_response_threshold_event <- function(response_threshold) {
...
-    if (time != parameters[["min_time"]]) {
+    # turn closure and excess hospitalisation switch on
+    is_hosp_switch_on <- as.logical(
+      rlang::env_get(parameters[["mutables"]], "hosp_switch")
+    )
+    if (time != parameters[["min_time"]] && !is_hosp_switch_on) {
       rlang::env_bind(
         parameters[["mutables"]],
-        switch = 1.0,
+        switch = 1.0, hosp_switch = 1.0,
```
Comment (richfitz) [inline R/closure.R:28]: With the coercion to logical above, might it be clearer to store these as `TRUE` rather than `1.0`?

File: R/closure.R [hunk 2]
```diff
@@ -59,8 +63,25 @@ make_rt_end_event <- function() {
...
+      # check if hospitalisations are greater than threshold
+      state <- array(
+        state,
+        c(N_AGE_GROUPS, N_EPI_COMPARTMENTS, N_ECON_STRATA)
+      )
+      total_hosp <- get_hospitalisations(state)
+
+      if (total_hosp > parameters[["hospital_capacity"]]) {
+        rlang::env_poke(
+          parameters[["mutables"]], "hosp_switch", 1.0
```
Comment (richfitz) [inline R/closure.R:76]: If you take the above suggestion, this becomes

```suggestion
          parameters[["mutables"]], "hosp_switch", TRUE
```

and similarly below, plus a simple change in `prepare_mutable_parameters` (and possibly elsewhere!)

File: data-raw/country_data.R [hunk 3]
```diff
@@ -306,6 +306,38 @@ to_remove <- c("Hong Kong", "Taiwan")
...
+
+# check names match
+stopifnot(
+  "Country data and hospital capacity names must match" =
+    identical(names(country_data), names(hospital_capacity))
+)
+
+country_data <- Map(country_data, hospital_capacity,
+  f = function(x, y) {
+    x[["hospital_capacity"]] <- round(y * sum(x[["demography"]]) / 1000)
```
Comment (richfitz) [inline data-raw/country_data.R:335]: we have some numbers in units of thousands of people, others in millions - at some point we might want to tidy that up, but best left until we are dealing with upstream sources I expect

**Review by richfitz — APPROVED** [review]
LGTM, though you might want to make that switch logical directly unless you have future plans for it that require it to be real valued

File: R/closure.R (hunk 1 above)
Comment (pratikunterwegs) [reply R/closure.R:28]: Sure - alternatively I can reduce the type conversion by writing the mortality rate calculation as

```r
increase = 0.6 for total of 1.6
omega = omega * (1 + hosp_switch * increase)
```

File: data-raw/country_data.R (hunk 3 above)
Comment (pratikunterwegs) [reply data-raw/country_data.R:335]: Agreed - I think this has to do with the original data sources and maybe conventions in how these quantities are represented? I've tried to keep the raw data files as close to original as possible so the processing steps are clear and version controlled.

Comment (richfitz) [reply data-raw/country_data.R:335]: yeah, it's just a comment for us to think about in future - working with things as raw as you can definitely seems the right move

Comment (pratikunterwegs) [reply data-raw/country_data.R:335]: Sure - in the long term, I wonder whether it's worth hooking the data source APIs - e.g. World Bank - to update these values regularly (or at least worth more than the trouble of doing this).

File: R/closure.R (hunk 1 above)
Comment (pratikunterwegs) [reply R/closure.R]: I've changed the switches to logicals instead of numerics for consistency - merging once checks pass.

**Comment by pratikunterwegs** [comment]
Both switches changed to logicals instead of numerics - merging now as checks pass.

**Pull request merged**